pkg/config/analysis/local/istiod_analyze.go

```diff
@@ -448,6 +448,11 @@ FilterMessages:
 			continue FilterMessages
 		}
 
+		// Filter out any messages that are generated from file templates
+		if m.Resource != nil && m.Resource.Origin.IsFromFile() {
+			continue FilterMessages
```

Comment:
Why would there be file based messages in istiod?

Reply:
This piece of code is used for both Istiod and Istioctl analyze. When we create fake pods from file templates, they may generate additional pod error/warning messages, and we need to filter them out in case of duplicated messages.